scripts/hydrogen_bond_propensity/hydrogen_bond_propensity_report.py

```diff
@@ -64,6 +66,56 @@ def make_diagram(mol, directory):
     return fname
 
 
+def make_diagram_components(crystal, molecule, directory, docx_template):
+    # Generates a diagram for each component in a given structure, with D/A atoms labelled if component has a donor/acceptor
+    molecule_diagram_generator = DiagramGenerator()
+    molecule_diagram_generator.settings.line_width = 1.6
+    molecule_diagram_generator.settings.font_size = 10
+    molecule_diagram_generator.settings.image_height = 300
+    molecule_diagram_generator.settings.shrink_symbols = False
+
+    asymmetric_molecule = crystal.asymmetric_unit_molecule
+    non_ch_molecule_no = 0
+    component_diagrams = []
+    donor_acceptor_check = []
+
+    # If no Z and Z' information (e.g. mol2), 'None' is returned for crystal.z_prime
+    z_prime_flag = False
+    if crystal.z_prime:
+        z_prime_flag = True
+
+    if z_prime_flag and crystal.z_prime < 1:
+        for m in molecule.components:  # generates an image for every molecule
+            atoms = m.atoms
+            donor_acceptor_check = [atom for atom in atoms if atom.is_donor or atom.is_acceptor]
+
+            if donor_acceptor_check:
+                lbl_da_atoms = [atom for atom in atoms if atom.is_donor or atom.is_acceptor]
+                img = molecule_diagram_generator.image(m, highlight_atoms=None, label_atoms=lbl_da_atoms)
+                non_ch_molecule_no += 1
+                fname = str(os.path.join(directory, '%s_component_%s.png' % (molecule.identifier, non_ch_molecule_no)))
+                if img:
+                    img.save(fname)
+                non_ch_img = add_picture_subdoc(fname, docx_template)
+                component_diagrams.append(non_ch_img)
+    else:
+        for m in asymmetric_molecule.components:  # generates an image for every molecule in the asymmetric unit
+            atoms = m.atoms
+            donor_acceptor_check = [atom for atom in atoms if atom.is_donor or atom.is_acceptor]
+
+            if donor_acceptor_check:
+                lbl_da_atoms = [atom for atom in atoms if atom.is_donor or atom.is_acceptor]
+                img = molecule_diagram_generator.image(m, highlight_atoms=None, label_atoms=lbl_da_atoms)
+                non_ch_molecule_no += 1
+                fname = str(os.path.join(directory, '%s_component_%s.png' % (molecule.identifier, non_ch_molecule_no)))
+                if img:
+                    img.save(fname)
+                non_ch_img = add_picture_subdoc(fname, docx_template)
+                component_diagrams.append(non_ch_img)
```

Comment:
This seems to do the same as above but for every component in the asymmetric molecule. It might be worth putting it in a function and calling it with the different molecules and donors atoms etc :)

Reply:
Put in a function to streamline